lightning/src/ln/chan_utils.rs

```diff
@@ -41,13 +42,19 @@ use core::ops::Deref;
 use crate::chain;
 use crate::util::crypto::sign;
 
-pub(crate) const MAX_HTLCS: u16 = 483;
-pub(crate) const OFFERED_HTLC_SCRIPT_WEIGHT: usize = 133;
-pub(crate) const OFFERED_HTLC_SCRIPT_WEIGHT_ANCHORS: usize = 136;
-// The weight of `accepted_htlc_script` can vary in function of its CLTV argument value. We define a
-// range that encompasses both its non-anchors and anchors variants.
-pub(crate) const MIN_ACCEPTED_HTLC_SCRIPT_WEIGHT: usize = 136;
-pub(crate) const MAX_ACCEPTED_HTLC_SCRIPT_WEIGHT: usize = 143;
+/// Maximum number of one-way in-flight HTLC (protocol-level value).
+pub const MAX_HTLCS: u16 = 483;
+/// The weight of HTLC second-stage `offered_htlc_script` non-anchor variant.
+pub const OFFERED_HTLC_SCRIPT_WEIGHT: usize = 133;
+/// The weight of HTLC second-stage `offered_htlc_script` anchor variant.
+pub const OFFERED_HTLC_SCRIPT_WEIGHT_ANCHORS: usize = 136;
+
+/// The weight of `accepted_htlc_script` can vary in function of its CLTV argument value.
+/// We define a range that encompasses both its non-anchors and anchors variants.
+/// Lower bound of the range.
+pub const MIN_ACCEPTED_HTLC_SCRIPT_WEIGHT: usize = 136;
+/// Upper bound of the range.
+pub const MAX_ACCEPTED_HTLC_SCRIPT_WEIGHT: usize = 143;
```

Comment:
If you're an advanced lightning signer/client, you could sanitize the whole transaction field-by-field, including the witness against your max hardcoded upper bound sat/vb. Especially you would like to protect against fee extraction issue if your fee estimator oracle is broken.

I don't deny they're hard to use correctly. At the really same time, I think exporting the protocol-level values is aimed towards advanced upstream users who have a precise use-case in mind. This is different than misusing your payment sending methods, where you would like API users to avoid double-spending themselves, here I would agree.

Reply:
Fair, IIRC we accept things with height < 250 for functional tests CLTV arguments. I think one way could be to add the spec canonical `ACCEPTED_HTLC_SCRIPT_WEIGHT_LEGACY` and `ACCEPTED_HTLC_SCRIPT_WEIGHT_ANCHORS` from: https://github.com/lightning/bolts/blob/master/03-transactions.md#expected-weight-of-htlc-timeout-and-htlc-success-transactions even if we don't use them internally. What do you think ?

Reply:
I took your ugh as an argument. Exposed only the max value and `HTLCClaim::from_witness`, if this is what you meant.